Queued retry storage extension WAL

pmm-sumo opened the pull request:
Work in progress

bogdandrutu committed:
Revert machine type to ubuntu 16.04 to pass deb package test (#3271)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

eddyleelin committed:
Added doc.go files to the consumer package and subpackages (#3270)

* Added consumertest doc.go

* Added consumerhelper doc.go

* Fixed typo in consumerhelper doc.go

* Moved consumererror doc to doc.go

* Added consumer doc.go

* Added newline to end of doc.go

* Added simple doc.go

* Fixed typo in pdata doc.go

* Update CHANGELOG.md

bogdandrutu committed:
Revert PR#3128 (#3276)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

sethAmazon committed:
Add Error Writer For File Exporter Tests And Remove Limited Writer #3225 (#3275)

Error Writer Is Internal To File Exporter Test And Will No Be Apart Of Test Util Like Limited Writer

jpkrohling committed:
Add margin of error for TestBallastMemory (#3249)

Fixes #3233

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

sincejune committed:
hook up start and shutdown functions in fileexporter (#3260)

* hook up start and shutdown functions in fileexporter

* fix tests

* Revert "fix tests"

This reverts commit f296b1035c298e854941b982f49e831d9ee53101.

* check file closed

* Revert changes in exporter.go

* Update factory.go

Co-authored-by: Bogdan Drutu <lazy@splunk.com>

pavankrish123 committed:
Refactor otlphttp exporter for upcoming API chnages (#3289)

* Refactor otlphttp exporter for upcoming API chnages

* addressed review comments

pavankrish123 committed:
Refactored zipkin exporter for the upcoming ToClient confighttp API changes.  (#3290)

* Refactored zipkin exporter for the upcoming ToClient confighttp API changes

* added suggestion from the review

* added suggestion from the review

pavankrish123 committed:
Refactor otlp exporter to upcoming auth changes (#3288)

* refactor otlp exporter to upcoming auth changes

* addressed review comments

* addressed review comments

* addressed review comments

* addressed review comments

* added suggestion from the review

* added suggestion from the review

* added suggestion from the review

pmalek-sumo commented on exporter/exporterhelper/README.md:
```diff
@@ -17,12 +17,48 @@ The following configuration options can be modified:
   - `enabled` (default = true)
   - `num_consumers` (default = 10): Number of consumers that dequeue batches; ignored if `enabled` is `false`
-  - `queue_size` (default = 5000): Maximum number of batches kept in memory before data; ignored if `enabled` is `false`;
+  - `queue_size` (default = 5000): Maximum number of batches kept in memory before data; ignored if `enabled` is `false` or WAL is enabled;
```
> Maximum number of batches kept in memory before data;

before data? :)

pmalek-sumo commented on exporter/exporterhelper/README.md:
```diff
@@ -17,12 +17,48 @@ The following configuration options can be modified:
+
+### WAL
+
+When `wal_enabled` is set, the queue is being buffered to by the storage extension. This has some limitations currently,
```
```suggestion
When `wal_enabled` is set, the queue is being buffered to disk by the storage extension. This has some limitations currently,
```

pmalek-sumo commented on exporter/exporterhelper/README.md:
```diff
@@ -17,12 +17,48 @@ The following configuration options can be modified:
+### WAL
+
+When `wal_enabled` is set, the queue is being buffered to by the storage extension. This has some limitations currently,
+the items that are currently being handled by a consumer are not backed by the persistent storage, which means
+that in case of a sudden shutdown, they might be lost.
```
IMHO this needs a different wording. You mean that items that have been passed down to consumers are flushed/removed from the WAL hence will be lost in case of a sudden shutdown?

pmalek-sumo commented on exporter/exporterhelper/README.md:
```diff
@@ -17,12 +17,48 @@ The following configuration options can be modified:
+                         x     x     x        │
+ ┌─────────WAL-backed queue────x─────x───┐    │    ┌─Consumer #2─┐
+ │                       x     x     x   │    │    │    ┌───┐    │
+ │     ┌───┐     ┌───┐ ┌─x─┐ ┌─x─┐ ┌─x─┐ │    │    │    │ 2 │    ├───► Permanent
+ │ n+1 │ n │ ... │ 4 │ │ 3 │ │ 2 │ │ 1 │ ├────┼───►│    └───┘    │      failure
```
Maybe add a cross to the arrow next to Permanet Failure to suggest the data does not get out?

pmalek-sumo commented on exporter/exporterhelper/common.go:
```diff
@@ -52,8 +52,12 @@ type request interface {
 	onError(error) request
 	// Returns the count of spans/metric points or log records.
 	count() int
+	marshall() ([]byte, error)
```
```suggestion
	marshal() ([]byte, error)
```

pmalek-sumo commented on exporter/exporterhelper/consumers_queue.go:
```diff
@@ -0,0 +1,23 @@
+
+package exporterhelper
+
+// This is largely based on queue.BoundedQueue and matches the subset used in the collector
+type consumersQueue interface {
+	StartConsumers(num int, callback func(item interface{}))
+	Produce(item interface{}) bool
+	Stop()
+	Size() int
+}
```
Do you mind adding a bit more verbose description to `consumersQueue` and other methods, especially `StartConsumers`

pmalek-sumo commented on exporter/exporterhelper/logs.go:
```diff
@@ -55,6 +68,10 @@ func (req *logsRequest) export(ctx context.Context) error {
 	return req.pusher(ctx, req.ld)
 }
 
+func (req *logsRequest) marshall() ([]byte, error) {
```
```suggestion
func (req *logsRequest) marshal() ([]byte, error) {
```

and accordingly elsewhere

pmalek-sumo commented on exporter/exporterhelper/wal.go:
```diff
@@ -0,0 +1,305 @@
+	return err == nil
+}
+
+// Stop stops accepting items and shuts-down the queue and closes the WAL
```
nitpick:

```suggestion
// Stop stops accepting items, shuts down the queue and closes the WAL.
```

pmm-sumo replied:
aaaand 🤦  - look kids, this is what late night coding produces :)

pmalek-sumo commented on exporter/exporterhelper/wal.go:
```diff
@@ -0,0 +1,305 @@
+}
+
+func uint64ToBytes(val uint64) ([]byte, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, binary.LittleEndian, val)
```
Escape analysis might optimize this away but you can explicitly put it on stack:

```suggestion
	var buf bytes.Buffer
	err := binary.Write(&buf, binary.LittleEndian, val)
```

pmm-sumo replied:
Oh, sweet!

bogdandrutu committed:
Fix duplicate the usages (#3283)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Fix oc to pdata translation for sum non-monotonic cumulative (#3272)

Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/3266

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

IrisTuntun committed:
Move BigEndian helper functions in tracetranslator to an internal package  (#3298)

* Move BigEndian helper functions in  to an internal package

* revert changelog

* update changelog

* Change to idutils and add doc.go

* Update internal/idutils/doc.go

Co-authored-by: Anthony Mirabella <a9@aneurysm9.com>

* Update CHANGELOG.md

Co-authored-by: Anthony Mirabella <a9@aneurysm9.com>

carlosalberto committed:
Remove typo in zpagesextension's README. (#3307)

bogdandrutu committed:
Fix error in TestBallast <Elements should be the same type> (#3309)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove unnecessary public func testutil.TempSocketName (#3291)

* Remove unnecessary public func testutil.TempSocketName

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

* Update changelog

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

mxiamxia committed:
remove legacy grpc receiver endpoint for GA release (#3256)

* remove legacy grpc receiver endpoint for GA release

* Update factory.go

Co-authored-by: Bogdan Drutu <lazy@splunk.com>
Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

euniceek committed:
Edit Type/Function Documentation in the Config Package (#3296)

* Fix typos in Config Package

* Ensure comments are above

* Address merge conflicts and additional typos

* Address merge conflicts

* Address documentation errors

* Change validateconfig to validate

tydhot committed:
fix bug  issue#3118 and issue#3161 (#3310)

euniceek committed:
Add doc.go files to the Config package (#3284)

* Add overview and move to seperate godoc

* Move config overview to doc.go

mxiamxia committed:
Hide OpenCensus reference from public APIs in obsreport package (#3253)

pavankrish123 committed:
Refactored PromRemoteWriteExporter to create client in Start to accomodate future update to ToClient() apis  (#3308)

* Refactored PromRemoteWriteExporter to create client in Start

* fixed impi errors

* Fixed per review comments

bogdandrutu committed:
Move config.Parser to configparser.Parser, preparation for ParserProvider interface (#3304)

Idea is that we will eventually move the https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/parserprovider/provider.go#L25 in this package, so everything about config parsing is in this package.

Signed-off-by: Bogdan Drutu bogdandrutu@gmail.com

humivo committed:
Remove deprecated funcs (#3314)

bogdandrutu committed:
Add otlpjson as a serializer implementation (#3238)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Rename internal/otlp files, avoid duplicate initialization (#3317)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Change Marshaler/Unmarshaler to be interfaces (#3318)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove obsreport.GRPCServerWithObservabilityEnabled, enable observability in config (#3315)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Add interested folks in Prom to codeowners, so they get notified about reviews in Prom (#3319)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove obsreport.ProcessorMetricViews, use BuildProcessorCustomMetricName where needed (#3316)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

pavankrish123 committed:
Refactored Jaeger exporter to move connection into start for upcoming Auth changes. (#3299)

* Refactored Jaeger exporter to move connection into start

* added more thorough cleznup

* fixed linting

bogdandrutu committed:
Rename TraceData to Traces in tests (#3325)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove Receive from obsreport.Receiver funcs (#3326)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Rename internal funcs push*Data to push* (#3324)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Rename configtest.LoadConfigFile to configtest.LoadConfigAndValidate (#3306)

Add a new configtest.LoadConfig to simplify testing errors on Validate.

Signed-off-by: Bogdan Drutu bogdandrutu@gmail.com

jpkrohling committed:
Rename OIDC auth extension (#3247)

This PR renames the OIDC Auth Extension to oidcauthextension, following the naming pattern agreed as part of #3128

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

bogdandrutu committed:
Remove Export from obsreport.Exporter funcs (#3333)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

pavankrish123 committed:
Port client side authentication interfaces from PR #3128  (#3321)

This PR ports the client authenticator interfaces from PR #3128  in a piece meal fashion. The interfaces are currently not used by any client configurations and are published for reviewing only. 

Plan is to modify ToClient and ToDialOptions apis once all the clients are prepared per #3287 (core and contrib side as well).

Link to tracking Issue:
#3287 #3282

Testing:
Unit tests, [manual test described (for only oidc)](https://github.com/open-telemetry/opentelemetry-collector/pull/3128#issuecomment-842485897)

dependabot[bot] committed:
Bump github.com/google/go-cmp from 0.5.5 to 0.5.6 (#3341)

Bumps [github.com/google/go-cmp](https://github.com/google/go-cmp) from 0.5.5 to 0.5.6.
- [Release notes](https://github.com/google/go-cmp/releases)
- [Commits](https://github.com/google/go-cmp/compare/v0.5.5...v0.5.6)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pavankrish123 committed:
Refactored OpenCensus exporter to move connection into start for upcoming Auth changes. (#3339)

Updated OpenCensus for upcoming client auth extension changes

Link to tracking Issue:
#3287 #3282

Commit:
Fixing a minor bug in the README: The log receivers were not sorted alphabetically, fixed it.

dependabot[bot] committed:
Bump github.com/cenkalti/backoff/v4 from 4.1.0 to 4.1.1 (#3343)

Bumps [github.com/cenkalti/backoff/v4](https://github.com/cenkalti/backoff) from 4.1.0 to 4.1.1.
- [Release notes](https://github.com/cenkalti/backoff/releases)
- [Commits](https://github.com/cenkalti/backoff/compare/v4.1.0...v4.1.1)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump go.uber.org/zap from 1.16.0 to 1.17.0 (#3342)

Bumps [go.uber.org/zap](https://github.com/uber-go/zap) from 1.16.0 to 1.17.0.
- [Release notes](https://github.com/uber-go/zap/releases)
- [Changelog](https://github.com/uber-go/zap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/uber-go/zap/compare/v1.16.0...v1.17.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump go.uber.org/zap from 1.16.0 to 1.17.0 in /cmd/issuegenerator (#3344)

Bumps [go.uber.org/zap](https://github.com/uber-go/zap) from 1.16.0 to 1.17.0.
- [Release notes](https://github.com/uber-go/zap/releases)
- [Changelog](https://github.com/uber-go/zap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/uber-go/zap/compare/v1.16.0...v1.17.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

andresbeckruiz committed:
Edited `make gendependabot` Task to Enable Dependabot for GitHub Actions (#3312)

* enabled dependabot for github actions by editing make gendependabot task in MakeFile

* run make gendependabot command

* Added changelog entry

* fixed changelog entry spacing

pmatyjasek-sumo committed:
Standarize Settings, Params and Parameters in Extensions (#3294)

* Replace ExtensionCreateParams with ExtensionCreateSettings.
Replace all dependencies in Extensions.

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

* Update changelog

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

# Conflicts:
#	CHANGELOG.md

* Add missing files

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 1 to 2.2.3 (#3349)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 1 to 2.2.3.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v1...v2.2.3)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/shirou/gopsutil (#3350)

Bumps [github.com/shirou/gopsutil](https://github.com/shirou/gopsutil) from 3.21.4+incompatible to 3.21.5+incompatible.
- [Release notes](https://github.com/shirou/gopsutil/releases)
- [Commits](https://github.com/shirou/gopsutil/compare/v3.21.4...v3.21.5)

---
updated-dependencies:
- dependency-name: github.com/shirou/gopsutil
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bryanuribe committed:
Review component package grammar (#3303)

* fixed grammar in processor.go

* fixed grammar in receiver.go

* fixed spelling

* modified component.go

* modified exporter.go

* fixed host.go

* fixed grammar in processor.go

* fixed grammar in component.go

* fixed grammar in exporter.go

* fixed grammar in extension.go

* fixed format error

* fixed lint errors in component.go

cole-easton committed:
Automate triggering of doc-update on release (#3330)

* Automate triggering of doc-update on release

* Create changelog entry

Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

pmatyjasek-sumo committed:
Replace ExporterCreateParams with ExporterCreateSettings. (#3164)

Replace all dependencies in Exporters.

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

# Conflicts:
#	exporter/prometheusremotewriteexporter/factory.go

# Conflicts:
#	exporter/fileexporter/factory.go
#	exporter/opencensusexporter/factory_test.go
#	exporter/otlpexporter/factory_test.go
#	exporter/prometheusremotewriteexporter/factory_test.go

# Conflicts:
#	CHANGELOG.md
#	exporter/kafkaexporter/factory.go
#	exporter/opencensusexporter/factory.go
#	exporter/otlpexporter/factory.go
#	exporter/otlphttpexporter/factory.go
#	exporter/zipkinexporter/factory.go

# Conflicts:
#	CHANGELOG.md
#	component/exporter.go
#	exporter/opencensusexporter/factory_test.go

pmatyjasek-sumo committed:
Standarize Settings, Params and Parameters in Receiver (#3167)

* Replace ReceiverCreateParams with ReceiverCreateSettings.
Replace all dependencies in Receivers.

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

# Conflicts:
#	component/receiver.go

# Conflicts:
#	exporter/opencensusexporter/opencensus_test.go
#	exporter/prometheusexporter/end_to_end_test.go

* Update changelog

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

# Conflicts:
#	CHANGELOG.md

# Conflicts:
#	CHANGELOG.md

# Conflicts:
#	CHANGELOG.md

* Add missing file

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

pmatyjasek-sumo committed:
Standarize Settings, Params and Parameters in Processors (#3181)

* Replace ProcessorCreateParams with ProcessorCreateSettings.
Replace all dependencies in Processors.

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

* Update changelog

Signed-off-by: Patryk Matyjasek <pmatyjasek@sumologic.com>

# Conflicts:
#	CHANGELOG.md

# Conflicts:
#	CHANGELOG.md

Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

austinlparker committed:
Sync docs changes between website and repo

kisieland committed:
Make the batch processor limit data points rather than metrics. (#3141)

**Link to tracking Issue:** #2754

This change introduces a BenchmarkBatchMetricProcessor that stress tests batching logic.
Results before:
`BenchmarkBatchMetricProcessor-12           20000             80614 ns/op`
Results after the change:
`BenchmarkBatchMetricProcessor-12           20000             96184 ns/op`

tigrannajaryan committed:
Fix getting-started.md: change Linux to Windows (#3356)

bogdandrutu committed:
Remove deadcode from opencensus receiver (#3352)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Hide unnecessary public struct in obsreport (#3353)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Add Marshaler/Unmarshaler interfaces for Logs and Metrics (#3351)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

jpkrohling committed:
Increase max RAM for load test (#3365)

Fixes #2911

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

eddyleelin committed:
Edit documentation in cmd/pdatagen (and consumer/pdata) (#3361)

* Create snyk-container-analysis.yml

* Create snyk-container-analysis.yml

* Update base_slices.go

* Update log_structs.go

* Update metrics_structs.go

* Update resource_structs.go

* Update trace_structs.go

* Updated consumer/pdata/generated*.go files

* Delete snyk-container-analysis.yml

* Fixed formatting errors in pdata

Co-authored-by: Karen Xu <karenyrxu@gmail.com>
Co-authored-by: Iris Song <chengcs@amazon.com>

bogdandrutu committed:
Avoid exposing internal implementation public in OC/OTEL receivers (#3355)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

dmitryax committed:
Add telemetry for dropped data due to exporter sending queue overflow (#3328)

* Add telemetry for dropped data due to exporter sending queue overflow

This change adds internal metrics for dropped spans, metric points and log records when exporter sending queue is full:
- exporter/enqueue_failed_metric_points
- exporter/enqueue_failed_spans
- exporter/enqueue_failed_log_records

* Make report*EnqueueFailure methods private

By moving them to the package where they are being used. It requires some code duplication

pavankrish123 committed:
Updated configgrpc ToDialOptions and confighttp ToClient apis to take extensions configuration map  (#3340)

This PR is a port of the configfrpc's ToDialOptions() and confighttp ToClient() from PR #3128  in a piece meal fashion. 

The following are the changes
- Refactored configgrpc.PerRPCAuth as extension implementing configauth.GrpcClientAuthenticator
- Plugged in extensions configuration to all the grpc based clients in the core (OTLP, OpenCensus, Jaeger, JaegerReceiver)
- Plugged in extensions configuration to all the HTTP based clients in the core (Zipkin, OTLPHTTP)

Link to tracking Issue:
 #3282 #3276

Testing:
Unit tests, [manual test described (for only oidc)](#3128 (comment))

rmfitzpatrick committed:
Fix pipelines typo in config loader test config (#3359)

odeke-em committed:
receiver/prometheus/internal: add createNodeAndResourcePdata for Prometheus->OTLP Pdata (#3139)

* receiver/prometheus/internal: add createNodeAndResourcePdata for Prometheus->OTLP Pdata

Starts the progressive effort to convert directly from
    Prometheus->OTLPPdata

directly instead of

    Prometheus->OpenCensusProto->OTLPPdata

by adding a converter for node+resource -> pdata.Resource.

Updates #3137

* Use .Attributes to skip unsafe extraction plus use require.Equal

Addresses feedback pointed out by Bogdan, Tigran, Jaana
by using Resource.Attributes().Sort() instead and also avoid
using go-cmp/cmp and instead opt to using require.Equal.

* Add comment and rename parity test check

* Add createNodeAndResource unit test

sincejune committed:
Remove unnecessary functions in PrometheusRemoteWriteExporter (#3261)

* remove unnecessary functions

* rename functions

* update

bryanuribe committed:
Improve component package docs (#3305)

* added doc.go file for component package

* added description for package componenthelper

* fixed comment for Shutdown

* added default build info for readability in go docs

* added doc.go for componenterror

* added doc.go for componenthelper

* added newline

* fixed grammar in descriptions

* added newline in componenthelper/doc.go

* added an

* fixed format error

* updated component package comments

* addressing lint error

IrisTuntun committed:
Remove the proto dependency in goldendataset for traces (#3322)

* Remove the proto dependency in goldendataset for traces

* revert changelog

* update changelog

* Update consumer/pdata/common.go

Co-authored-by: Anthony Mirabella <a9@aneurysm9.com>

* Fix common.go where there is a missing "}"

* Add functions to convert proto keyvalues to AttributeMap

* further remove proto dependency

* Add PR number to CHANGELOG

* update resource_generator_test and resource_to_oc_test

* Return nil instead of empty objects for some generator functions.

Co-authored-by: Anthony Mirabella <a9@aneurysm9.com>
Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

punya committed:
Add punya as triager (#3376)

Relates to #3358.

bogdandrutu committed:
Fix usage of CopyTo/Append by passing the destination to the function calls (#3377)

Fixes couple of bugs from PR#3322, see comments in that PR.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Bump github.com/jaegertracing/jaeger from 1.22.0 to 1.23.0 (#3374)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Add componenttest.NewNop*CreateSettings to simplify tests (#3375)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Convert fileexporter to not depend on InternalRep (#3384)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove unnecessary public API in testbed (#3390)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

rmfitzpatrick committed:
Run make gotidy for check-contrib (#3392)

If there are core dependency changes that affect the contrib go module state the check-contrib make target will fail. These changes run make gotidy in the contrib project before running the tests (per https://github.com/open-telemetry/opentelemetry-collector/pull/3337/files#r647145882)

bogdandrutu committed:
Rename pdata files to match traces/metrics/logs (#3385)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Small nits in goldendataset for metrics file name and public funcs (#3380)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

euniceek committed:
Change grpc to GRPC (#3285)

This PR is one of the 3 PRs working to address the issue: Reviewing and Improving config package.
Main changes that were made in this PR was to ensure types and function names with `grpc` is consistent throughout. For exported functions and types, changes were made such that `GRPC` is used. As for unexported functions and types, `gRPC` is now used.

Addresses part of #3051

odeke-em committed:
receiver/prometheus/internal: direct metricfamily Prometheus->OTLP + equivalent tests (#3145)

Adds direct metricfamily Prometheus->OTLP conversion as well as equivalence tests.
After this change, there will be 2 other changes coming up and then finally we'll be able
to remove the dependency of OpenCensus in the Prometheus receiver.

Updates #3137

pmcollins committed:
Add support for markdown generation (#3100)

This change contains a CLI tool, called 'docsgen', that generates markdown files for collector
components. The markdown files present the configuration metadata extracted by the `configschema`
API in a human readable form that can be used to manually configure the collector.

This change also makes some modifications to the package formerly knows as `schemagen`, renaming
it to `configschema` and exporting some things, because it no longer generates a schema yaml file, but
rather provides an equivalent API used by docsgen.

Also, this PR includes one sample, generated .md document: `receiver/otlpreceiver/config.md`

jchengsfx committed:
Release v0.28.0 (#3395)

bogdandrutu committed:
Samll error message fix, remove reference to OC (#3381)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove batch counter from the testbed DataProvider, not used (#3391)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Hide testbed DataProvider implementation structs (#3389)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove usages of InternalRep when possible in pdata tests (#3404)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Convert otlptext to implement Marshaler interfaces (#3366)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove also the Nil vs Empty tests since in pdata they are equivalent. (#3399)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Updates https://github.com/open-telemetry/opentelemetry-collector/issues/3104

jrcamp committed:
zipkin translator small cleanup (#3411)

odeke-em committed:
receiver/prometheus: reject datapoints with duplicate label keys (#3408)

As of Prometheus 2.16.0 (released on 2020-02-13), datapoints with
duplicate label keys MUST be rejected and the Prometheus RemoteWrite
exporter tests were failing, but that was a red herring as the
real issue was really in the receiver.

Fixes https://github.com/open-telemetry/wg-prometheus/issues/44
Fixes #3407

bogdandrutu committed:
Add encoder/decoder and marshaler/unmarshaler for OTLP protobuf (#3401)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Use the new marshaler/unmarshaler in Kafka exporter (#3403)

* Add encoder/decoder and marshaler/unmarshaler for OTLP protobuf

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

* Use the new marshaler/unmarshaler in Kafka exporter

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

jrcamp committed:
Convert zipkinv2 to to/from translator interfaces (#3409)

* Convert zipkinv2 to to/from translator interfaces

Doesn't yet have encode/decoders.

* move v2 test files to zipkinv2

rakyll committed:
Adjust default retry settings for the PRW Exporter (#3416)

The default retry settings are far off from the Prometheus
Server's behavior. Retry more aggressively to avoid excessive
in-memory buffering.

Also fixes open-telemetry/wg-prometheus#54 because the test
times out before the retry is sent.

bogdandrutu committed:
Update FileDataProvider to use the new model.Unmarshaler, avoid InternalRep usage (#3398)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

mapno committed:
Use cmd's context instead of background (#3418)

bogdandrutu committed:
Add tests for OTLP JSON encode/decode (#3400)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

pmm-sumo committed:
Copy contrib's storage interface to core

jrcamp committed:
zipkinv1: move to translator and encoders interfaces (#3419)

* zipkinv1: move to translator and encoders interfaces

* review feedback

bogdandrutu committed:
Remove unnecessary proto marshaler tests, avoid using InternalRep in otlprecv (#3386)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Use the new marshaler/unmarshaler in Kafka receiver (#3402)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Avoid compiling in parallel same time without that (#3382)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove test only code, avoid unsing proto and InternalRep in testdata (#3387)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

rmfitzpatrick committed:
Replace Viper with Koanf to add map case sensitivity (#3337)

dependabot[bot] committed:
Bump go.uber.org/atomic from 1.7.0 to 1.8.0 (#3429)

Bumps [go.uber.org/atomic](https://github.com/uber-go/atomic) from 1.7.0 to 1.8.0.
- [Release notes](https://github.com/uber-go/atomic/releases)
- [Changelog](https://github.com/uber-go/atomic/blob/master/CHANGELOG.md)
- [Commits](https://github.com/uber-go/atomic/compare/v1.7.0...v1.8.0)

---
updated-dependencies:
- dependency-name: go.uber.org/atomic
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bogdandrutu committed:
Remove usage of grpc-gateway in otlpreceiver. Http protocols are different than grpc (#3422)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.10.0 to 1.11.0 (#3428)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.10.0 to 1.11.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.10.0...v1.11.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bogdandrutu committed:
Change oltpreceiver to use the new pdatagrpc, avoid proto dependency (#3406)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

eddyleelin committed:
Edit type/func documentation in the consumer package (#3269)

* Edited grammar of pdata doc.go

* Added missing word

* Fixed typos in pdata/common.go

* Added missing String method docs

This fix makes the pdata folder consistent in documenting the String methods.

* Fixed spacing in doc example

* Fixed spacing issues in multiple examples

This issue was causing many example snippets to show up incorrectly on the godoc

* Fixed spacing issue on Range example

* Fixed spacing in At example

* Fixed more spacing errors in examples

* Fixed error in ResourceMetrics header describing InstrumentationLibraryMetrics

* Fixed another header referencing wrong method

* Added documentation for pdata.Resource

* Updated outdated links to Span OTLP definition

* Cleaned up wording in SpanStatus doc

* Fixed more spacing errors in examples

* Fixed typos in type docs

* Cleaned up formatting in consumererror

* Cleaned up consumerhelper doc typos

* Updated docs in consumertest

- Fixed wrong description of NewErr
- Added doc for Capabilities method

* Update CHANGELOG.md

* Fixed formatting errors in cmd/pdatagen

Changing here will ensure that the changes to consumer/pdata/generated_* files will persist after running `go run cmd/pdatagen/main.go`.

* Removed unused imports

* Update CHANGELOG.md

* Fixed generated code lint failures

* Revert changes in pdatagen and pdata/generated

* Update NewErr docstring

For func defined in consumer/consumertest/err.go

Co-authored-by: Tigran Najaryan <4194920+tigrannajaryan@users.noreply.github.com>

* Updated docstring for TraceState

Renamed "message" to "string"

Co-authored-by: Tigran Najaryan <4194920+tigrannajaryan@users.noreply.github.com>

bogdandrutu committed:
Remove usage of internal proto in configgrpc tests (#3435)

Updates #3104

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Use the new Marshaler in the OTLP HTTP exporter (#3433)

Updates #3104

Signed-off-by: Bogdan Drutu bogdandrutu@gmail.com

bogdandrutu committed:
Remove unnecessary public structs from testbed (#3432)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Move zipkin constants to an internal package (#3431)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Add grpc response struct for all signals instead of returning interface (#3437)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Fix build due to a race condition in merging PRs (#3440)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Add comments to translation interfaces, clarify on error return value cannot be used (#3421)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove usage of ToOtlpProtoBytes, will be removed after Marshaler is released (#3434)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

jrcamp committed:
zipkinv2: add encoders, decoders, marshalers (#3426)

* zipkinv2: add encoders, decoders, marshalers

Also update zipkin receiver to use them.

* don't check returned value on error

* review feedback

Co-authored-by: Bogdan Drutu <bogdandrutu@gmail.com>

rakyll committed:
Return concrete error type (#3360)

Go development practices favors returning concrete types over interfaces, so the users don't have to force cast the return value to the concrete type and the APIs self document the returned error type.

bogdandrutu committed:
Small cleanups in testbed (#3445)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Remove unnecessary exclude rules from golangci (#3441)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

dashpole committed:
Disallow renaming metrics using the prometheus receiver (#3410)

* disallow renaming metrics using the prometheus receiver

* add TODO for removing the validation

pmm-sumo committed:
Basic persistent queue implementation based on storage extension interface

pmm-sumo committed:
Code cleanup and clarifications

Co-authored-by: Patryk Małek <pmalek@sumologic.com>

pmm-sumo committed:
Replace atomic with channel

Co-authored-by: Patryk Małek <pmalek@sumologic.com>
